control-plane-operator/controllers/hostedcontrolplane/v2/dnsoperator/component.go

```diff
@@ -0,0 +1,49 @@
+package dnsoperator
+
+import (
+	oapiv2 "github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/v2/oapi"
+	component "github.com/openshift/hypershift/support/controlplane-component"
+	"github.com/openshift/hypershift/support/util"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	ComponentName = "dns-operator"
+)
+
+var _ component.ComponentOptions = &dnsOperator{}
+
+type dnsOperator struct {
+}
+
+// IsRequestServing implements controlplanecomponent.ComponentOptions.
+func (d *dnsOperator) IsRequestServing() bool {
+	return false
+}
+
+// MultiZoneSpread implements controlplanecomponent.ComponentOptions.
+func (d *dnsOperator) MultiZoneSpread() bool {
+	return true
+}
+
+// NeedsManagementKASAccess implements controlplanecomponent.ComponentOptions.
+func (d *dnsOperator) NeedsManagementKASAccess() bool {
+	return false
+}
+
+func NewComponent() component.ControlPlaneComponent {
+	return component.NewDeploymentComponent(ComponentName, &dnsOperator{}).
+		WithAdaptFunction(adaptDeployment).
+		WithManifestAdapter(
+			"kubeconfig.yaml",
+			component.WithAdaptFunction(adaptKubeconfigSecret),
+		).
+		WithDependencies(oapiv2.ComponentName).
+		InjectAvailabilityProberContainer(util.AvailabilityProberOpts{
+			KubeconfigVolumeName: "kubeconfig",
+			RequiredAPIs: []schema.GroupVersionKind{
+				{Group: "operator.openshift.io", Version: "v1", Kind: "DNS"},
```

Comment:
do we not need this option anymore?
https://github.com/openshift/hypershift/blob/2753dcc8cfb7591a61d64d26766526f844cebffa/control-plane-operator/controllers/hostedcontrolplane/dnsoperator/dnsoperator.go#L165